core/kazoo_number_manager-1.0.0/src/knm_phone_number.erl

```diff
@@ -0,0 +1,549 @@
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec feature(number(), ne_binary()) -> ne_binary().
+feature(Number, Feature) ->
+    wh_json:get_value(Feature, Number#number.features).
+
+-spec set_feature(number(), ne_binary(), _) -> number().
+set_feature(N, Feature, Data) ->
+    Features = features(N),
+    N#number{features=wh_json:set_value(Feature, Data, Features)}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec state(number()) -> ne_binary().
+state(Number) ->
+    Number#number.state.
+
+-spec set_state(number(), ne_binary()) -> number().
+set_state(N, State) ->
+    N#number{state=State}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec reserve_history(number()) -> ne_binaries().
+reserve_history(Number) ->
+    Number#number.reserve_history.
+
+-spec set_reserve_history(number(), ne_binaries()) -> number().
+set_reserve_history(N, History) ->
+    N#number{reserve_history=History}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec ported_in(number()) -> boolean().
+ported_in(Number) ->
+    Number#number.ported_in.
+
+-spec set_ported_in(number(), boolean()) -> number().
+set_ported_in(N, Ported) ->
+    N#number{ported_in=Ported}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec module_name(number()) -> ne_binary().
+module_name(Number) ->
+    Number#number.module_name.
+
+-spec set_module_name(number(), ne_binary()) -> number().
+set_module_name(N, Name) ->
+    N#number{module_name=Name}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec region(number()) -> ne_binary().
+region(Number) ->
+    Number#number.region.
+
+-spec set_region(number(), ne_binary()) -> number().
+set_region(N, Region) ->
+    N#number{region=Region}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec auth_by(number()) -> ne_binary().
+auth_by(Number) ->
+    Number#number.auth_by.
+
+-spec set_auth_by(number(), ne_binary()) -> number().
+set_auth_by(N, AuthBy) ->
+    N#number{auth_by=AuthBy}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec dry_run(number()) -> ne_binary().
+dry_run(Number) ->
```

Comment:
Why create temporary variables in these? Just pattern match the record's field and return it. No reason to bind Number.